server/datapool/service.go

```diff
@@ -18,4 +18,5 @@ func RegisterHandlers(svc *service.Service, router *mux.Router) {
 
 	router.HandleFunc("/v0/data-pool/pools/{poolId}/rounds/{round}/data", s.handleValidateData).Methods(http.MethodPost)
 	router.HandleFunc("/v0/data-pool/pools/{poolId}/data", s.handleDownloadData).Methods(http.MethodGet)
+	router.HandleFunc("/v0/data-pool/pools/register/validator", s.handleRegisterDataValidator).Methods(http.MethodPost)
```

Comment:
I think we don't need to create this API because this operation shouldn't be executed by users. Instead, I thought we can execute the `MsgRegisterDataValidator` tx automatically when the datavald is started, only if it's not already registered. Then, the problem is that we don't know which `endpoint` should be contains in the tx. In my opinion, we can add the `public-endpoint` in the `[http]` section of `config.toml`.

Reply:
If so, you can also delete the `server/datapool/register_data_validator.go` file.